clicommand/agent_start.go

```diff
@@ -811,8 +815,29 @@ func shutdownHook(log logger.Logger, hooksPath string) {
 		log.Error("Failed to create shell object for shutdown hook: %v", err)
 		return
 	}
+
+	// pipe from hook output to logger
+	r, w := io.Pipe()
+	sh.Logger = &shell.WriterLogger{Writer: w, Ansi: !cfg.NoColor} // for Promptf
+	sh.Writer = w                                                  // for stdout+stderr
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		scan := bufio.NewScanner(r) // log each line separately
+		log = log.WithFields(logger.StringField("hook", "shutdown"))
+		for scan.Scan() {
+			log.Info(scan.Text())
```

Comment:
Unfortunately there's no way to differentiate stdout vs stderr without substantial changes to our `shell`.